### ISSUE TITLE
what are the supported events?

### ISSUE THREAD

**Issue by tcstory** [request/request#1682]
my english is poor.

i wanna know what the event that request supports,and the event order.

**Comment by Freyert** [comment]
I have . . .
1. `response`: Emitted when an HTTP response is received.
2. `data`: Emitted each time data is received.
3. `end`: Emitted when a connection ends.

**Comment by tcstory** [comment]
@Freyert why the documentation doesn't give the events, i think this is very useful

**Comment by simov** [comment]
:+1: for improved docs. Most of the events are inherited from their respective modules around the `http(s)` modules. You can check out the official *nodejs/iojs* docs as well.

**Comment by bradvogel** [comment]
+1 can someone document these in the readme?

**Comment by Freyert** [comment]
Yeah, I think this needs more documentation as the Node docs for HTTP Client events itself is also based on knowing the Object hierarchy; knowing the events for streams, tcp sockets, and http. I'm sure HTTPS also has more.

Tentatively, I'd love to ponder this. I'll see if it's feasible for me.

**Comment by LoicMahieu** [comment]
https://iojs.org/api/http.html#http_class_http_server

**Comment by wearhere** [comment]
+1 Also it would be great to know in what order the events should be fired, and what events if any are shortcircuited if you call `abort`.

**Comment by Freyert** [comment]
@LoicMahieu is that part of the convergence as well?

**Comment by LoicMahieu** [comment]
@Freyert Sorry I don't know what you mean.

**Comment by Freyert** [comment]
@LoicMahieu I meant, do you know if those events are in node and io, as well as the upcoming convergence of IO and Node.

**Comment by LoicMahieu** [comment]
@Freyert Sorry I don't have a list of those events but it would be great... Little googling for that gives nothing... There is a lake of documentation about which events are trigged, when and in which order...

**Comment by simov** [comment]
I think the only request specific event is `redirect` (correct me if I'm wrong). Once request is migrated to use streams2, the events will be exactly the same as in the node/iojs docs (and currently they should be pretty close to that, if not the same).

**Comment by ohenepee** [comment]
What's the difference between the `end` and `close` events?
